vortex-layout/src/reader.rs

```diff
@@ -1,35 +1,21 @@
 use std::sync::Arc;
 
-use vortex_array::ArrayData;
 use vortex_dtype::DType;
-use vortex_error::VortexResult;
-use vortex_expr::ExprRef;
+use vortex_scan::AsyncEvaluator;
 
-use crate::operations::Operation;
-use crate::{LayoutData, RowMask};
-
-pub type EvalOp = Box<dyn Operation<Output = ArrayData>>;
+use crate::LayoutData;
 
 /// A [`LayoutReader`] is an instance of a [`LayoutData`] that can cache state across multiple
 /// operations.
-pub trait LayoutReader {
+///
+/// Since different row ranges of the reader may be evaluated by different threads, it is required
+/// to be both `Send` and `Sync`.
+pub trait LayoutReader: Send + Sync + AsyncEvaluator {
     /// Returns the [`LayoutData`] of this reader.
     fn layout(&self) -> &LayoutData;
 
-    /// Creates a new evaluator for the layout. It is expected that the evaluator makes use of
-    /// shared state from the [`LayoutReader`] for caching and other optimisations.
-    //
-    // NOTE(ngates): we have chosen a general "run this expression" API instead of  separate
-    //  `filter(row_mask, expr) -> row_mask` + `project(row_mask, field_mask)` APIs.
-    //  The reason for this is so we can eventually support cell-level push-down.
-    //  If we only projected using a field mask, then it means we need to download all the data
-    //  for the rows of field present in the row mask. When I say cell-level push-down, I mean
-    //  we can slice the cell directly out of storage using an API like
-    //  `SegmentReader::read(segment_id, byte_range: Range<usize>)`. This is a highly advanced
-    //  use-case, but can prove invaluable for large cell values such as images and video.
-    //  If instead we make the projection API `project(row_mask, expr)`, then identical to the
-    //  filter API and there's now no point having two. Hence: `evaluate(row_mask, expr)`.
-    fn create_evaluator(self: Arc<Self>, row_mask: RowMask, expr: ExprRef) -> VortexResult<EvalOp>;
+    /// Returns the [`AsyncEvaluator`] for this reader.
+    fn evaluator(&self) -> &dyn AsyncEvaluator;
```

Comment:
If LayoutReader implements AsyncEvaluator, is there any reason to also have this method?